src/utils.rs

```diff
@@ -0,0 +1,190 @@
+use std::{
+    env::Args,
+    fs,
+    io::{Error, Read, Write, stdin},
+    os::fd::AsRawFd,
+    process::exit,
+};
+
+use termios::{ECHO, ICANON, TCSANOW, Termios, tcsetattr};
+
+use crate::{
+    error::VMError,
+    hardware::{CondFlag, Memory, Register, Registers},
+};
+
+/// Extends a number represented in 'bit_count' bits into
+/// 16 bits, always taking into account the sign of
+/// the original number
+pub fn sign_extend(mut x: u16, bit_count: usize) -> Result<u16, VMError> {
+    // Get MSB and check if it is a 1
+    let bitcount_sub = bit_count
+        .checked_sub(1)
+        .ok_or(VMError::Arithmetic(String::from(
+            "Underflow when substracting",
+        )))?;
+    let msb = x >> bitcount_sub;
+    if msb != 0 {
+        // If the MSB is 1 it means it is negative, else its positive
+        x |= 0xFFFF << bit_count;
+    }
+    Ok(x)
+}
+
+/// It updates the register COND where we have the condition flag
```

Comment:
To keep consistency with the rest of the documentation:

```suggestion
/// Updates the register COND where we have the condition flag
```